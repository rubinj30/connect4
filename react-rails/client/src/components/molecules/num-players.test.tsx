import React from 'react';
import { shallow } from 'enzyme';
import { NumPlayers } from './num-players.component';

describe('NumPlayers', () => {
    it('should render properly', () => {
        const fixture = (
            <NumPlayers isCompTurn={'n'} changeCompTurn={jest.fn()} />
        );
        const result = shallow(fixture);

        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
    it('changeTurn should be called on click', () => {
        const mockFunc = jest.fn();
        const fixture = (
            <NumPlayers isCompTurn={'n'} changeCompTurn={mockFunc} />
        );
        const result = shallow(fixture);
        expect(mockFunc).toHaveBeenCalledTimes(0);
        result.simulate('click');
        expect(mockFunc).toHaveBeenCalledTimes(1);
    });
});
