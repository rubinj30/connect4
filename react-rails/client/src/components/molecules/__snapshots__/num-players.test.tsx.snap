// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPlayers should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NumPlayers
    changeCompTurn={[MockFunction]}
    isCompTurn="n"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="pr1"
        >
          Is computer on?
        </span>,
        <span
          className="pa2 ba b--blue option "
        >
          No
        </span>,
        <span
          className="pa2 ba option bg-blue white bw2 b--red"
        >
          Yes
        </span>,
      ],
      "className": "br2",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Is computer on?",
          "className": "pr1",
        },
        "ref": null,
        "rendered": "Is computer on?",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No",
          "className": "pa2 ba b--blue option ",
        },
        "ref": null,
        "rendered": "No",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Yes",
          "className": "pa2 ba option bg-blue white bw2 b--red",
        },
        "ref": null,
        "rendered": "Yes",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="pr1"
          >
            Is computer on?
          </span>,
          <span
            className="pa2 ba b--blue option "
          >
            No
          </span>,
          <span
            className="pa2 ba option bg-blue white bw2 b--red"
          >
            Yes
          </span>,
        ],
        "className": "br2",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Is computer on?",
            "className": "pr1",
          },
          "ref": null,
          "rendered": "Is computer on?",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No",
            "className": "pa2 ba b--blue option ",
          },
          "ref": null,
          "rendered": "No",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Yes",
            "className": "pa2 ba option bg-blue white bw2 b--red",
          },
          "ref": null,
          "rendered": "Yes",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
